dashboard/app/helpers/thank_donors_interstitial_helper.rb

```diff
@@ -0,0 +1,14 @@
+module ThankDonorsInterstitialHelper
+  # Determine whether or not to show the thank donors interstitial popup to a user
+  # We only show the interstitial to signed in users on their first login,
+  # and also do not want to show it when it might conflict with other dialogs.
+  def self.show?(user, request)
+    return false if user.nil?
+    return false unless user.sign_in_count == 1
+    return false if user.age.nil? # Age dialog
+    return false if user && user.teacher? && !user.accepted_latest_terms? # Terms of Service dialog
```

Comment:
given that we have an increasing number of dialogs, I'd be interested to see these conditions also pulled out into their own helpers, and possibly also some kind of general "DialogHelper" that can gather all the different dialog conditions into one place.

That could probably be follow-up work, though.

Reply:
Yes, totally agree.